pkg/asset/installconfig/aws/regions.go

```diff
@@ -1,32 +1,43 @@
 package aws
 
 import (
-	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"context"
 
-	"github.com/openshift/installer/pkg/rhcos"
-	"github.com/openshift/installer/pkg/types"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/sets"
 )
 
 // knownPublicRegions is the subset of public AWS regions where RHEL CoreOS images are published.
 // This subset does not include supported regions which are found in other partitions, such as us-gov-east-1.
-// Returns: a map of region identifier to region description.
-func knownPublicRegions(architecture types.Architecture) map[string]string {
-	required := rhcos.AMIRegions(architecture)
+// Returns: a list of region names.
+func knownPublicRegions() []string {
+	regions := []string{}
+	cfg, err := config.LoadDefaultConfig(context.Background(), config.WithDefaultRegion("us-east-1"))
+	if err != nil {
+		logrus.Debugf("failed to create aws config: %v", err)
+		return regions
+	}
+	ec2Config := ec2.NewFromConfig(cfg)
 
-	regions := make(map[string]string)
-	for _, region := range endpoints.AwsPartition().Regions() {
-		if required.Has(region.ID()) {
-			regions[region.ID()] = region.Description()
-		}
+	output, err := ec2Config.DescribeRegions(context.Background(), &ec2.DescribeRegionsInput{
+		AllRegions: aws.Bool(true),
```

Comment:
So the difference with `AllRegions` is that it will show opt-in regions that haven't been opted in to, right? :thinking: My gut feeling is we don't want to set this to true.